### DISCUSSION TITLE
It is possible to replace (probably) unsafe C code ?

### DISCUSSION THREAD

**Discussion by ghost** [Kunzisoft/KeePassDX#1690] [Q&A] [answered]
As the title says. Are there some libraries written in safe languages which provide the same functionality (cryptography if I understood correctly) ?

**Comment by J-Jamet** [comment]
Is it possible to replace a car engine that works well, does what it's supposed to do, has no particular problems with a more modern engine that has other constraints, must perform as well as the old one (because had been optimized on purpose) and will have to be fully adapted because (on a guess without having looked) there may be unsafe code?
Of course. Good luck.

**Reply by J-Jamet** [reply, marked as answer]
No kidding, I'd already tried replacing the engine with a Go version, which works without a hitch, but performance is reduced on KDF algorithms.

**Reply by ghost** [reply]
I understand. Yes, as you said, I did not actually read the code. I was asking just out of curiosity, hence the “Q&A category”. I unfortunately don’t have the abilities to make “a new engine”.  Anyway, thank you for you answer, and of course thank you for you for project. :)
